### PR TITLE
#507 - refactored ApiChangeUserPassword.java

### PR DESCRIPTION
**Pull request by olenagerasimova**
Part of #507 - refactored `ApiChangeUserPassword` to use `Credentials` interface and its implementations to change user password.

### CODE REVIEW

File: src/main/java/com/artipie/api/ApiChangeUserPassword.java [hunk 1]
```diff
@@ -87,50 +79,25 @@ public Response response(final String line,
...
+            Single.just(body).to(ContentAs.STRING).map(
+                encoded -> URLEncodedUtils.parse(encoded, StandardCharsets.UTF_8)
+                    .stream()
+                    .filter(pair -> pair.getName().equals("password"))
+                    .map(NameValuePair::getValue)
+                    .findFirst().orElseThrow()
+            ).flatMapCompletable(
+                pass -> Completable.fromFuture(
+                    this.settings.credentials().thenCompose(
+                        cred -> cred.add(user, pass, Credentials.PasswordFormat.SHA256)
```
Comment (olegmoz) [inline src/main/java/com/artipie/api/ApiChangeUserPassword.java:92]: @olenagerasimova it seems that `pass` is stored in plain no matter what `PasswordFormat` is. It looks like a bug

File: src/main/java/com/artipie/api/ApiChangeUserPassword.java [hunk 2]
```diff
@@ -87,50 +79,25 @@ public Response response(final String line,
...
-                .toSingleDefault(
-                    new RsWithHeaders(
-                        new RsWithStatus(RsStatus.FOUND),
-                        new Header("Location", String.format("/dashboard/%s", user))
-                    )
-                ).to(SingleInterop.get())
+            ).toSingleDefault(
+                new RsWithHeaders(
+                    new RsWithStatus(RsStatus.FOUND),
+                    new Header("Location", String.format("/dashboard/%s", user))
```
Comment (olegmoz) [inline src/main/java/com/artipie/api/ApiChangeUserPassword.java:98]: @olenagerasimova there is standard `Location` header class you may use here

File: src/main/java/com/artipie/api/artifactory/AddUpdateUserSlice.java [hunk 3]
```diff
@@ -74,8 +75,9 @@ public Response response(final String line, final Iterable<Map.Entry<String, Str
                         passw -> passw.map(
                             haspassw -> this.settings.credentials()
                                 .thenCompose(
-                                    cred -> cred.add(username, haspassw)
-                                        .thenApply(ok -> new RsWithStatus(RsStatus.OK)))
+                                    cred -> cred.add(
+                                        username, haspassw, Credentials.PasswordFormat.PLAIN
```
Comment (olegmoz) [inline src/main/java/com/artipie/api/artifactory/AddUpdateUserSlice.java:79]: @olenagerasimova I see that passwords were stored were stored in plain format before these changes, but maybe we should stick to sha256 format by default?

**Review by olegmoz — CHANGES_REQUESTED** [review]
@olenagerasimova thanks, please see the comments

File: src/main/java/com/artipie/api/ApiChangeUserPassword.java (hunk 1 above)
Comment (olenagerasimova) [reply src/main/java/com/artipie/api/ApiChangeUserPassword.java:92]: @olegmoz yes, you are right

**Comment by olenagerasimova** [comment]
@olegmoz thanks, corrected

File: src/main/java/com/artipie/api/ApiChangeUserPassword.java [hunk 4]
```diff
@@ -89,13 +89,13 @@ public Response response(final String line,
             ).flatMapCompletable(
                 pass -> Completable.fromFuture(
                     this.settings.credentials().thenCompose(
-                        cred -> cred.add(user, pass, Credentials.PasswordFormat.SHA256)
+                        cred -> cred.add(user, pass, Credentials.PasswordFormat.PLAIN)
```
Comment (olegmoz) [inline src/main/java/com/artipie/api/ApiChangeUserPassword.java:92]: @olenagerasimova I am sorry, but why did you change format to `PLAIN` here?

**Review by olegmoz — CHANGES_REQUESTED** [review]
@olenagerasimova thanks, please see one question

**Comment by olenagerasimova** [comment]
@olegmoz I've misunderstood your previous comment, fixed

**Review by olegmoz — APPROVED** [review]
@olenagerasimova thanks, merging

**Comment by 0crat** [comment]
Job `gh:artipie/artipie#539` is not assigned, can't get performer
